src/Http/Http.Extensions/src/DescriptionAttribute.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http.Metadata;
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// Specifies a description for the endpoint in <see cref="Endpoint.Metadata"/>.
+/// </summary>
+/// <remarks>
+/// The OpenAPI specification supports a description attribute on operations and parameters that
+/// can be used to annotate endpoints with detailed, multiline descriptors of their behavior.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Method | AttributeTargets.Delegate, Inherited = false, AllowMultiple = false)]
+public sealed class DescriptionAttribute : Attribute, IDescriptionMetadata
```

Comment:
We might want to think thru these names in API review. The namespace Microsoft.AspNetCore.Http is imported by default and these names seem very generic. Perhaps ApiDescriptionAttribute or EndpointDescriptionAttribute?

Reply:
`EndpointDescription` seems sensible to use here since we have prior art with things like `EndpointName`.